src/modules/flow/twitter/twitter.c

```diff
@@ -0,0 +1,617 @@
+        &params, twitter_request_finished, cb_data->node);
+
+err:
+    sol_http_param_free(&params);
+    return connection;
+}
+
+static void
+digest_ready_cb(void *data, struct sol_message_digest *handle, struct sol_blob *output)
+{
+    int r;
+    char *authorization_header, *escaped_signature;
+    struct callback_data *cb_data = data;
+    struct sol_http_client_connection *connection;
+    struct sol_buffer buffer = SOL_BUFFER_INIT_EMPTY;
+    struct sol_buffer encode_buf;
+    struct twitter_data *mdata = sol_flow_node_get_private_data(cb_data->node);
+
+    r = sol_buffer_append_as_base64(&buffer, sol_str_slice_from_blob(output), SOL_BASE64_MAP);
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    r = sol_http_encode_slice(&encode_buf, sol_str_slice_from_str((char *)buffer.data));
+    SOL_INT_CHECK_GOTO(r, < 0, err_escape);
+    escaped_signature = sol_str_slice_to_string(sol_buffer_get_slice(&encode_buf));
+    sol_buffer_fini(&encode_buf);
+    SOL_NULL_CHECK_GOTO(escaped_signature, err_escape);
+
+    r = asprintf(&authorization_header, "OAuth "
+        "oauth_consumer_key=\"%s\", "
+        "oauth_nonce=\"%s\", "
+        "oauth_signature=\"%s\", "
+        "oauth_signature_method=\"HMAC-SHA1\", "
+        "oauth_timestamp=\"%s\", "
+        "oauth_token=\"%s\", "
+        "oauth_version=\"1.0\"",
+        mdata->consumer_key, cb_data->nonce, escaped_signature,
+        cb_data->timestamp, mdata->token);
+    SOL_INT_CHECK_GOTO(r, < 0, err_header);
+
+    connection = cb_data->cb(cb_data, authorization_header);
+    SOL_NULL_CHECK_GOTO(connection, err_connection);
+
+    r = sol_ptr_vector_append(&mdata->pending_conns, connection);
+    SOL_INT_CHECK_GOTO(r, < 0, err_append);
+
+    goto err_connection;
+
+err_append:
+    sol_http_client_connection_cancel(connection);
+err_connection:
+    free(authorization_header);
+err_header:
+    free(escaped_signature);
+err_escape:
+    sol_buffer_fini(&buffer);
+err:
+    free(cb_data->timestamp);
+    free(cb_data->status);
+    free(cb_data->nonce);
+    free(cb_data);
+}
+
+static int
+post_status(struct sol_flow_node *node, const char *status)
+{
+    int r = -EINVAL;
+    struct sol_blob *blob;
+    struct callback_data *cb_data;
+    struct sol_message_digest *digest;
+    struct sol_buffer buf;
+    struct twitter_data *mdata = sol_flow_node_get_private_data(node);
+    char *signature, *params, *escaped_params, *escaped_status;
+    struct sol_message_digest_config *digest_config = &(struct sol_message_digest_config) {
+        .api_version = SOL_MESSAGE_DIGEST_CONFIG_API_VERSION,
+        .algorithm = "hmac(sha1)",
+        .on_digest_ready = digest_ready_cb,
+    };
+
+    cb_data = calloc(1, sizeof(*cb_data));
+    SOL_NULL_CHECK(cb_data, -ENOMEM);
+
+    cb_data->node = node;
+    cb_data->cb = post_request;
+
+    cb_data->nonce = generate_nonce();
+    SOL_NULL_CHECK_GOTO(cb_data->nonce, err);
+
+    r = asprintf(&cb_data->timestamp, "%li", (long int)time(NULL));
```

Comment:
You never need to use the timestamp as string alone, all usages I saw is to feed it to `asprintf()`,  thus you don't nee to give it %s, use %li instead.